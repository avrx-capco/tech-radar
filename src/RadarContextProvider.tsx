import { createContext, useState, FC } from 'react';

export type RadarContextType = {
  category: string;
  technology: string;

  setCategory: (newCategory: string) => void;
  setTechnology: (newComptenecy: string) => void;
};

export const RadarContext = createContext<RadarContextType>({
  category: '',
  technology: '',
  setCategory: (name: string) => {},
  setTechnology: (name: string) => {},
});

export const RadarContextProvider: FC = ({ children }) => {
  const [category, setCategory] = useState('');
  const [technology, setTechnology] = useState('');
  const initialValue = {
    category,
    technology,
    setCategory,
    setTechnology,
  };

  return (
    <RadarContext.Provider value={initialValue}>
      {children}
    </RadarContext.Provider>
  );
};
