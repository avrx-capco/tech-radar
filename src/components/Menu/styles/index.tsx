import { Link } from "react-router-dom";
import styled from "styled-components/macro";
import Drawer from "@material-ui/core/Drawer";
import AppBar from "@material-ui/core/AppBar";
import Grid from "@material-ui/core/Grid";

export const StyledAppBar = styled(AppBar)`
  && {
    background: transparent;
    padding-top: 20px;
    box-shadow: none;
    #purpose {
      :hover {
        text-decoration: underline;
      }
    }
  }
`;

export const LogoGridItem = styled(Grid)`
  && {
    z-index: 1400;
    margin-left: 60px;

    div {
      color: ${({ theme }) => theme.pink};
      font-weight: 700;
      font-size: 24px;
    }

    @media screen and (max-width: 1000px) {
      margin-left: 10px;
      img {
        height: 36px;
      }
      div {
        font-size: 19px;
      }
    }
  }
`;

export const StyledDrawer = styled(Drawer)`
  div.MuiPaper-root {
    width: 40%;
    background-color: #0e0330;
    color: white;
    @media screen and (max-width: 768px) {
      width: 100%;
    }
  }
  .MuiIconButton-root {
    margin: 20px;
    float: right;
  }
  ul {
    height: ${window.innerHeight}px;
    width: 100%;
    padding: 150px 0 0 80px;
    background: radial-gradient(rgba(230, 35, 109, 0.7) 0%, transparent 90%);
    a {
      font-size: 30px;
      font-weight: bold;
      color: inherit;
      :hover {
        text-decoration: underline;
      }
      @media screen and (max-width: 768px) {
        display: inline-block;
        text-align: center;
      }
    }
    @media screen and (max-width: 1000px) {
      padding: 100px 0 0 20px;
      background: radial-gradient(rgba(230, 35, 109, 0.7) 0%, transparent 150%);
      .MuiListItem-root {
        padding-bottom: 0;
      }
    }
  }
`;

export const StyledLink = styled(Link)`
  display: inline-block;
  color: white;
  font-size: 20px;
  margin-right: 60px;
  :hover {
    color: white;
  }
  img {
    margin-right: 10px;
  }
  @media screen and (max-width: 1000px) {
    display: none;
  }
`;
