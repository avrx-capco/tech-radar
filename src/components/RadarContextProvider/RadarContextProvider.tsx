/* eslint-disable @typescript-eslint/no-empty-function */
import { createContext, useState, FC } from "react";

export type RadarContextType = {
  category: string;
  technology: string;
  setCategory: (newCategory: string) => void;
  setTechnology: (newComptenecy: string) => void;
};

export const RadarContext = createContext<RadarContextType>({
  category: "",
  technology: "",
  setCategory: () => {},
  setTechnology: () => {},
});

export const RadarContextProvider: FC = ({ children }) => {
  const [category, setCategory] = useState("");
  const [technology, setTechnology] = useState("");
  const initialValue = {
    category,
    technology,
    setCategory,
    setTechnology,
  };

  return (
    <RadarContext.Provider value={initialValue}>
      {children}
    </RadarContext.Provider>
  );
};
