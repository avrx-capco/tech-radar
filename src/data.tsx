export const radial = [
  {
    name: 'DEVOPS',
    preferred: [
      { name: 'GitHub', link: 'https://img.icons8.com/fluent/452/github.png' },
      {
        name: 'CircleCI',
        link: 'https://img.icons8.com/color/452/circleci.png',
      },
      {
        name: 'Terraform',
        link:
          'https://symbols.getvecta.com/stencil_97/45_terraform-icon.d8dd637866.svg',
      },
      {
        name: 'Valut',
        link:
          'https://symbols.getvecta.com/stencil_100/11_vault-icon.469c5366cc.svg',
      },
      { name: 'SonarQube', link: 'https://pbs.twimg.com/profile_images/1224335491899760641/h404B5dU_400x400.jpg' },
      {
        name: 'Elastic',
        link:
          'https://symbols.getvecta.com/stencil_79/34_elastic-search-icon.bdaf397325.svg',
      },
    ],
    skilled: [
      { name: 'Bitbucket', link: 'https://cdn.worldvectorlogo.com/logos/bitbucket-icon.svg' },
      {
        name: 'Jenkins',
        link:
          'https://symbols.getvecta.com/stencil_97/67_the-jenkins-project-icon.0141f64fe7.svg',
      },
      { name: 'TeamCity', link: 'https://cdn.worldvectorlogo.com/logos/teamcity-icon.svg' },
      {
        name: 'Istio',
        link:
          'https://symbols.getvecta.com/stencil_84/70_istio-icon.9d92202138.svg',
      },
      {
        name: 'Ansible',
        link:
          'https://symbols.getvecta.com/stencil_73/122_ansible-icon.e1db432c74.svg',
      },
      {
        name: 'Packer',
        link:
          'https://symbols.getvecta.com/stencil_91/3_packer-icon.53584ee287.svg',
      },
    ],
    scaling: [
      {
        name: 'Twistlock',
        link:
          'https://symbols.getvecta.com/stencil_98/71_twistlock-icon.8db1b813dd.svg',
      },
      { name: 'Kiali', link: 'https://design.jboss.org/kiali/logo/final/PNG/kiali_icon_darkbkg_1280px.png' },
      { name: 'ConcourseCl', link: 'https://cdn.worldvectorlogo.com/logos/concourse-1.svg' },
      {
        name: 'Consul',
        link:
          'https://symbols.getvecta.com/stencil_77/63_consul-icon.0452948395.svg',
      },
    ],
  },
  {
    name: 'DATABASES',
    preferred: [
      {
        name: 'Postgres',
        link: 'https://img.icons8.com/color/452/postgreesql.png',
      },
      {
        name: 'Neo4J',
        link:
          'https://symbols.getvecta.com/stencil_261/28_neo4j.404372b29e.svg',
      },
      {
        name: 'MongoDB',
        link: 'https://www.iamsteveni.com/static/img/mongodb.png',
      },
    ],
    skilled: [
      {
        name: 'MySQL',
        link: 'https://www.iamsteveni.com/static/img/mysql.png',
      },
      {
        name: 'Oracle',
        link: 'https://img.icons8.com/color/452/oracle-logo.png',
      },
      {
        name: 'SQL Server',
        link: 'https://img.icons8.com/color/452/microsoft-sql-server.png',
      },
    ],
    scaling: [
      {
        name: 'BigQuery',
        link:
          'https://symbols.getvecta.com/stencil_75/69_bigquery-icon.eb9185e8d2.svg',
      },
    ],
  },
  {
    name: 'HOSTING',
    preferred: [
      {
        name: 'Kubernetes',
        link: 'https://img.icons8.com/color/452/kubernetes.png',
      },
      {
        name: 'Serverless',
        link:
          'https://symbols.getvecta.com/stencil_95/69_serverless-icon.b1abb9b824.svg',
      },
      {
        name: 'OpenShift',
        link:
          'https://symbols.getvecta.com/stencil_90/47_openshift-icon.e98513fd69.svg',
      },
    ],
    skilled: [{ name: 'PCF', link: 'https://avatars.githubusercontent.com/u/5497370?s=200&v=4' }],
    scaling: [{ name: 'PKS', link: 'https://blogs.vmware.com/virtualblocks/files/2018/08/PKS-2.png' }],
  },
  {
    name: 'CLOUD',
    preferred: [
      {
        name: 'GCP',
        link: 'https://img.icons8.com/color/452/google-cloud.png',
      },
    ],
    skilled: [
      {
        name: 'AWS',
        link: 'https://img.icons8.com/color/452/amazon-web-services.png',
      },
      { name: 'Azure', link: 'https://img.icons8.com/color/452/azure-1.png' },
    ],
    scaling: [
      {
        name: 'Digital Ocean',
        link:
          'https://symbols.getvecta.com/stencil_78/61_digital-ocean-icon.0d013f7d26.svg',
      },
    ],
  },
  {
    name: 'INTEGRATION',
    preferred: [
      {
        name: 'Rest',
        link: 'https://miro.medium.com/max/300/1*1RDFnS8FgAOQFegtuynxWw.png',
      },
      { name: 'Camel', link: 'https://camel.apache.org/_/img/logo-d.svg' },
      { name: 'gRPC', link: 'https://docs.konghq.com/assets/images/icons/hub/kong-inc_grpc-gateway.png' },
      {
        name: 'Kafka',
        link:
          'https://symbols.getvecta.com/stencil_74/14_apache-kafka-icon.96e46bbe3a.svg',
      },
    ],
    skilled: [
      {
        name: 'ActiveM',
        link:
          'https://symbols.getvecta.com/stencil_73/17_activemq-icon.66dcff47ad.svg',
      },
      {
        name: 'IBM MQ',
        link:
          'https://anypoint.mulesoft.com/exchange/organizations/68ef9520-24e9-4cf2-b2f5-620025690913/assets/com.mulesoft.connectors/mule-ibm-mq-connector/icon/svg/?sha=97c7ca8fde33f64f5da710f17de072de73f67b11',
      },
    ],
    scaling: [
      {
        name: 'GraphQL',
        link:
          'https://symbols.getvecta.com/stencil_82/66_graphql-icon.291fc05a38.svg',
      },
    ],
  },
  {
    name: 'BACKEND',
    preferred: [
      {
        name: 'Java',
        link: 'https://img.icons8.com/color/452/java-coffee-cup-logo.png',
      },
      {
        name: 'Node',
        link: 'https://www.iamsteveni.com/static/img/nodejs.png',
      },
      { name: 'Go', link: 'https://img.icons8.com/color/344/golang.png' },
    ],
    skilled: [{ name: 'Python', link: 'https://img.icons8.com/color/452/python.png' }],
    scaling: [{ name: 'Kotlin', link: 'https://img.icons8.com/color/344/kotlin.png' }],
  },
  {
    name: 'FRONTEND',
    preferred: [
      {
        name: 'React',
        link: 'https://img.icons8.com/color/452/react-native.png',
      },
      {
        name: 'Angular',
        link: 'https://img.icons8.com/color/452/angularjs.png',
      },
    ],
    skilled: [],
    scaling: [
      { name: 'Flutter', link: 'https://img.icons8.com/color/452/flutter.png' },
      { name: 'Vue', link: 'https://img.icons8.com/color/344/vue-js.png' },
    ],
  },
  {
    name: 'MOBILE',
    preferred: [
      {
        name: 'React Native',
        link: 'https://img.icons8.com/color/452/react-native.png',
      },
    ],
    skilled: [
      { name: 'Swift', link: 'https://img.icons8.com/fluent/452/swift.png' },
      { name: 'Kotlin', link: 'https://img.icons8.com/color/344/kotlin.png' },
      { name: 'Java', link: 'https://img.icons8.com/color/452/java-coffee-cup-logo.png' },
      { name: 'Ionic', link: 'https://img.icons8.com/ios-filled/452/ionic.png' },
    ],
    scaling: [{ name: 'Flutter', link: 'https://img.icons8.com/color/452/flutter.png' }],
  },
];
// occurence can be 100,50,30
// 100 = preferred
// 50 = skilled
// scaling = scaling
export const bubble = [
  {
    name: 'BackEnd',
    data: [
      {
        tech: 'Java',
        occurence: 100,
        link: 'https://img.icons8.com/color/452/java-coffee-cup-logo.png',
      },
      {
        tech: 'Node',
        occurence: 100,
        link: 'https://img.icons8.com/color/344/nodejs.png',
      },
      {
        tech: 'Go',
        occurence: 100,
        link: 'https://img.icons8.com/color/344/golang.png',
      },
      {
        tech: 'Python',
        occurence: 50,
        link: 'https://img.icons8.com/color/452/python.png',
      },
      {
        tech: 'Kotlin',
        occurence: 30,
        link: 'https://img.icons8.com/color/344/kotlin.png',
      },
    ],
  },
  {
    name: 'FrontEnd',
    data: [
      {
        tech: 'React',
        occurence: 100,
        link: 'https://img.icons8.com/color/452/react-native.png',
      },
      {
        tech: 'Angular',
        occurence: 100,
        link: 'https://img.icons8.com/color/452/angularjs.png',
      },
      {
        tech: 'Vue',
        occurence: 30,
        link: 'https://img.icons8.com/color/344/vue-js.png',
      },
      {
        tech: 'Flutter',
        occurence: 30,
        link: 'https://img.icons8.com/color/452/flutter.png',
      },
    ],
  },
  {
    name: 'Mobile',
    data: [
      {
        tech: 'React Native',
        occurence: 100,
        link: 'https://img.icons8.com/color/452/react-native.png',
      },
      {
        tech: 'Swift',
        occurence: 50,
        link: 'https://img.icons8.com/fluent/452/swift.png',
      },
      {
        tech: 'Kotlin',
        occurence: 50,
        link: 'https://img.icons8.com/color/344/kotlin.png',
      },
      {
        tech: 'Java',
        occurence: 50,
        link: 'https://img.icons8.com/color/452/java-coffee-cup-logo.png',
      },
      {
        tech: 'Ionic',
        occurence: 50,
        link: 'https://img.icons8.com/ios-filled/452/ionic.png',
      },
      {
        tech: 'Flutter',
        occurence: 30,
        link: 'https://img.icons8.com/color/452/flutter.png',
      },
    ],
  },
];
