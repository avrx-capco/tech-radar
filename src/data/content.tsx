import { ImageAspectRatioOutlined } from '@material-ui/icons';

export type TechContentType = {
  technology: string;
  intro: string;
  content: Array<{
    name: string;
    intro: string;
    data: Array<{ name: string; description: string }>;
  }>;
  examples: Array<{ name: string; description: string }>;
};

export default [
  {
    technology: 'React',
    intro:
      'React is the world’s most popular JavaScript library that uses components to change content of a single-page Mobile and Web application without refreshing.',
    content: [
      {
        name: 'State Management',
        intro:
          'Managing states is one of the biggest challenges in large web applications. Using React alone may not be sufficient to handle such complexity. We need a way to share data across components by creating a central store that holds all states of your application.',
        data: [
          {
            name: 'Redux',
            description:
              'Redux provides a central store for states that each component can access.  Main building parts: store, actions, reducers and optional middleware.',
          },
          {
            name: 'ContextAPI',
            description:
              'Context provides a way to pass data through the component tree without having to pass props down manually at every level and helps to avoid the prop-drilling problem. 3 main building parts: context, provider and consumer.',
          },
        ],
      },
      {
        name: 'UI Libraries',
        intro: '',
        data: [
          {
            name: 'MaterialUI',
            description:
              'Google’s Material Design is a popular React UI library with responsive animations and it has been adopted by the wider design community.',
          },
          {
            name: 'SemanticUI',
            description:
              'It contains pre-built semantic components that helps create responsive layouts.',
          },
          {
            name: 'Ant Design of React',
            description:
              'Contains a set of high-quality components and demos for building rich, interactive user interfaces.',
          },
        ],
      },
      {
        name: 'Styling',
        intro: '',
        data: [
          {
            name: 'CSS and SASS',
            description:
              'This involves using separate stylesheets like our conventional way of styling our HTML websites either with CSS or a CSS preprocessor called SASS.',
          },
          {
            name: 'BEM - Block Element Modifier',
            description:
              "BEM It's a CSS naming convention for writing cleaner and more readable CSS classes. BEM aims to write independent CSS blocks increasing reusability.",
          },
          {
            name: 'Styled Components',
            description:
              'styled-components is a library for React and React Native that allows you to use component-level styles in your application that are written with a mixture of JavaScript and CSS using a technique called CSS-in-JS.',
          },
        ],
      },
    ],
    examples: [
      {
        name: 'Digital Engineering Framework App',
        description: '',
      },
      {
        name: 'UK Challenge App',
        description: '',
      },
    ],
  },
];
