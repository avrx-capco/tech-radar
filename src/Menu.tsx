import React, { useState, useContext } from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Divider from '@material-ui/core/Divider';
import ClearIcon from '@material-ui/icons/Clear';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';
import styled from 'styled-components/macro';
import logo from './logo.png';
import { ListItem } from '@material-ui/core';
import RadarContext from './RadarContext';

const useStyles = makeStyles((theme: any) => ({
  appBar: {
    background: 'black',
    paddingTop: 20,
  },
  list: {
    width: 250,
    margin: 20,
    fontSize: 20,
  },
  menuButton: {
    marginRight: theme.spacing(2),
    right: 0,
    float: 'right',
    fontSize: 'large',
  },
  title: {
    fontSize: 45,
  },
}));

export default function Header() {
  const classes = useStyles();
  const [open, setOpen] = useState(false);
  const { name, setName } = useContext(RadarContext);

  const handleClick = (item: string) => {
    setName(item);
  };

  const toggleDrawer = (open: boolean) => (event: any) => {
    if (
      event.type === 'keydown' &&
      (event.key === 'Tab' || event.key === 'Shift')
    ) {
      return;
    }
    setOpen(open);
  };

  const MenuItems = () => (
    <div
      className={classes.list}
      role='presentation'
      onClick={toggleDrawer(false)}
      onKeyDown={toggleDrawer(false)}
    >
      <IconButton
        onClick={toggleDrawer(false)}
        edge='start'
        className={classes.menuButton}
        color='inherit'
        aria-label='clear'
      >
        <ClearIcon fontSize='large' />
      </IconButton>
      <List>
        <ListItem button component={Link} to='/'>
          Home
        </ListItem>
        <Divider />
        {[
          'Mobile',
          'FrontEnd',
          'BackEnd',
          'Integration',
          'Cloud',
          'Hosting',
          'Databases',
          'Devops',
        ].map((item: string) => (
          <ListItem
          key={`menu-${item}`}
            button
            component={Link}
            to='/info'
            onClick={() => handleClick(item)}
          >
            {item}
          </ListItem>
        ))}
         <Divider />
        {/* <ListItem button component={Link} to='/table'>
          Table
        </ListItem> */}
      </List>
    </div>
  );

  return (
    <>
      <AppBar className={classes.appBar} position='static'>
        <Toolbar>
          <Grid justify='space-between' container>
            <Grid item>
              <img
                height={40}
                src={logo}
                alt='Logo'
                style={{ margin: '1rem' }}
              />
            </Grid>

            <Grid item>
              <div>
                <IconButton
                  onClick={toggleDrawer(true)}
                  edge='start'
                  className={classes.menuButton}
                  color='inherit'
                  aria-label='menu'
                >
                  <MenuIcon fontSize='large' />
                </IconButton>
              </div>
            </Grid>
          </Grid>
        </Toolbar>
      </AppBar>
      <Drawer anchor={'right'} open={open} onClose={toggleDrawer(false)}>
        <MenuItems />
      </Drawer>
    </>
  );
}
