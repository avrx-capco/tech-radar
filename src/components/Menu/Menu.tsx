import { useState, useContext } from "react";
import { Link } from "react-router-dom";
import List from "@mui/material/List";
import IconButton from "@mui/material/IconButton";
import MenuIcon from "@mui/icons-material/Menu";
import ClearIcon from "@mui/icons-material/Clear";
import Toolbar from "@mui/material/Toolbar";
import Grid from "@mui/material/Grid";
import logo from "images/logo.png";
import robot from "images/robot.svg";
import { ListItem } from "@mui/material";
import { RadarContext } from "../RadarContextProvider/RadarContextProvider";
import { categoryList } from "data/data";
import {
  StyledAppBar,
  LogoGridItem,
  StyledDrawer,
  StyledLink,
} from "./styles/";

const Menu = () => {
  const [open, setOpen] = useState(false);
  const { setCategory } = useContext(RadarContext);

  const handleClick = (item: string) => {
    setCategory(item.toLowerCase());
  };

  const toggleDrawer = (open: boolean) => (event: any) => {
    if (
      event.type === "keydown" &&
      (event.key === "Tab" || event.key === "Shift")
    ) {
      return;
    }
    setOpen(open);
  };

  return (
    <>
      <StyledAppBar position="static">
        <Toolbar>
          <Grid justifyContent="space-between" alignItems="start" container>
            <LogoGridItem item>
              <Link to="/home">
                <img height={85} src={logo} alt="Large Logo" />
                <div>TECH RADAR {new Date().getFullYear()}</div>
              </Link>
            </LogoGridItem>

            <Grid item alignItems="center" sx={{ display: "flex" }}>
              {window.location.pathname !== "/" && (
                <StyledLink id="purpose" to="/">
                  <img src={robot} alt="robot" width={28} height={26} />
                  Radar Purpose
                </StyledLink>
              )}

              <IconButton
                onClick={toggleDrawer(true)}
                onKeyDown={toggleDrawer(true)}
                edge="start"
                color="inherit"
                aria-label="menu"
              >
                <MenuIcon fontSize="large" />
              </IconButton>
            </Grid>
          </Grid>
        </Toolbar>
      </StyledAppBar>

      <StyledDrawer anchor={"right"} open={open} onClose={toggleDrawer(false)}>
        <div
          role="presentation"
          onClick={toggleDrawer(false)}
          onKeyDown={toggleDrawer(false)}
        >
          <IconButton
            onClick={toggleDrawer(false)}
            edge="start"
            color="inherit"
            aria-label="clear"
          >
            <ClearIcon fontSize="large" />
          </IconButton>
          <List>
            {categoryList.map((name: string) => (
              <ListItem
                id={`menu${name.replace(/\s/g, "-")}`}
                key={`menu-${name}`}
                button
                component={Link}
                to={`/category/${name.replace(/\s/g, "-").toLowerCase()}`}
                onClick={() => handleClick(name)}
              >
                {name}
              </ListItem>
            ))}
          </List>
        </div>
      </StyledDrawer>
    </>
  );
};

export default Menu;
