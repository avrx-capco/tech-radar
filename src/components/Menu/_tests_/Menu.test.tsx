import * as React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Menu from "../Menu";
import { categoryList } from "../../../data/data";
import { BrowserRouter as Router } from "react-router-dom";
import { RadarContextProvider } from "../../RadarContextProvider/RadarContextProvider";

describe("Menu", () => {
  it("should render Menu component", () => {
    const { container } = render(
      <Router>
        <Menu />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should cick Mennu icon and check that the list of categories have color white", () => {
    render(
      <Router>
        <RadarContextProvider>
          <Menu />
        </RadarContextProvider>
      </Router>
    );

    fireEvent.click(screen.getByLabelText("menu"));
    categoryList.forEach((name: string) => {
      expect(screen.getByText(name)).toHaveStyle("color:inherit");
      expect(screen.getByText(name)).toBeVisible();
    });
  });

  it("should press Tab key to open Menu and check that clear icon is in focus", () => {
    render(
      <Router>
        <RadarContextProvider>
          <Menu />
        </RadarContextProvider>
      </Router>
    );

    fireEvent.keyDown(screen.getByLabelText("menu"));
    userEvent.tab();

    expect(screen.getByLabelText("clear")).toHaveFocus();
  });

  it("should open Menu and check that Mobile link has the correct href attribute", () => {
    render(
      <Router>
        <RadarContextProvider>
          <Menu />
        </RadarContextProvider>
      </Router>
    );

    fireEvent.click(screen.getByLabelText("menu"));
    fireEvent.click(screen.getByText("Mobile"));

    expect(screen.getByText("Mobile")).toHaveAttribute(
      "href",
      "/category/mobile"
    );
  });

  it("should press Tab key to open Menu and check that Mobile link is in focus", () => {
    render(
      <Router>
        <RadarContextProvider>
          <Menu />
        </RadarContextProvider>
      </Router>
    );

    fireEvent.keyDown(screen.getByLabelText("menu"));
    userEvent.tab();
    userEvent.tab();

    expect(screen.getByText("Mobile")).toHaveFocus();
  });
});
