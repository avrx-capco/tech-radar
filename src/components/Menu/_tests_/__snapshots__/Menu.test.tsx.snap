// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render Menu component 1`] = `
.c0.c0 {
  background: transparent;
  padding-top: 20px;
  box-shadow: none;
}

.c0.c0 #purpose:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1.c1 {
  z-index: 1400;
  margin-left: 60px;
}

.c1.c1 div {
  font-weight: 700;
  font-size: 24px;
}

@media screen and (max-width:1000px) {
  .c1.c1 {
    margin-left: 10px;
  }

  .c1.c1 img {
    height: 36px;
  }

  .c1.c1 div {
    font-size: 19px;
  }
}

@media screen and (max-width:768px) {

}

@media screen and (max-width:768px) {

}

@media screen and (max-width:1000px) {

}

<header
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic c0 css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1lmj5vd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item c1 css-13i4rnv-MuiGrid-root"
      >
        <a
          href="/home"
        >
          <img
            alt="Large Logo"
            height="85"
            src="logo.png"
          />
          <div>
            TECH RADAR 
            2023
          </div>
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-qh4ozh-MuiGrid-root"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</header>
`;
